x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/bucket/geogrid/GeoHexVisitor.java

```diff
@@ -297,21 +297,21 @@ private GeoRelation relateTriangle(
         }
 
         boolean within = false;
-        if (edgeIntersectsQuery(aX, aY, bX, bY, false)) {
+        if (edgeIntersectsQuery(aX, aY, bX, bY, ab)) {
```

Comment:
From other implementation, I am more tempted to set the flag to be always true. I am afraid of edge cases if all edges are false.